megaparse/api/app.py

```diff
@@ -84,18 +92,24 @@ async def parse_file(
     )
     try:
         parser = parser_builder.build(parser_config)
-        with tempfile.NamedTemporaryFile(
-            delete=False, suffix=f".{str(file.filename).split('.')[-1]}"
-        ) as temp_file:
-            temp_file.write(file.file.read())
-            megaparse = MegaParse(parser=parser)
-            result = await megaparse.aload(file_path=temp_file.name)
-            return {"message": "File parsed successfully", "result": result}
+
+        megaparse = MegaParse(parser=parser)
+        if not file.filename:
+            raise Exception("No filename provided")
+        _, extension = os.path.splitext(file.filename)
```

Comment:
At some point, it would be safer to use regex with a list of known, supported extensions

Reply:
This will be done I think in the parsers with a list of supported extensions!